crates/oxc_linter/src/rules/typescript/no_explicit_any.rs

```diff
@@ -0,0 +1,374 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+"typescript-eslint(no-explicit-any): Using any disables many type checking rules and is generally best used only as a last resort or when prototyping code"
+)]
+#[diagnostic(severity(warning))]
+struct NoExplicitAnyDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoExplicitAny {
+    pub ignore_rest_args: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow the any type.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// The any type in TypeScript is a dangerous "escape hatch" from the type system.
+    /// Using any disables many type checking rules and is generally best used only as a last
+    /// resort or when prototyping code. This rule reports on explicit uses of the any keyword
+    /// as a type annotation.
+    ///
+    /// ### Example
+    /// ```typescript
+    /// const age: any = 'seventeen';
+    /// const ages: any[] = ['seventeen'];
+    /// const ages: Array<any> = ['seventeen'];
+    /// ```
+    NoExplicitAny,
+    correctness
```

Comment:
should probably be `restriction`: 

https://github.com/manuschillerdev/oxc/blob/42b97fc270e2e1601d8f1c93379cda712cd94255/crates/oxc_linter/src/rule.rs#L42